### PR TITLE
added filter state in redux and a filterToggle action

### PR DESCRIPTION
**Pull request by eddiewang**
Working branch to test 0 SC filter.

### CODE REVIEW

File: plugins/Wallet/js/components/transactionlist.js [hunk 1]
```diff
@@ -2,7 +2,8 @@ import PropTypes from 'prop-types'
 import React from 'react'
 import { List } from 'immutable'
 
-const TransactionList = ({transactions, ntransactions, actions}) => {
+const TransactionList = ({ transactions, ntransactions, actions, filter }) => {
+	console.log(filter)
```
Comment (avahowell) [inline plugins/Wallet/js/components/transactionlist.js]: leftover logging

File: plugins/Wallet/js/components/transactionlist.js [hunk 2]
```diff
@@ -11,42 +12,104 @@ const TransactionList = ({transactions, ntransactions, actions}) => {
...
+					txn.transactionsums.totalMiner.round(4).toNumber().toLocaleString() +
+					' SC (miner) '
+			}
+			if (valueData === '') {
+				valueData = '0 SC'
+			}
+			if (filter) {
+				if (valueData !== '0 SC') {
+					return (
+						<tr key={key}>
```
Comment (avahowell) [inline plugins/Wallet/js/components/transactionlist.js]: This is not very DRY, why not just add a `.filter()` before this map:

```
transactionComponents.filter((txn) => {
  if (!filter) {
     return true
  }
  return txn.transactionsums.totalSiacoin.abs.gt(0) || txn.transactionsums.totalSiafund.abs().gt(0) || txn.transactionsums.totalMiner.abs().gt(0)
}).map((txn, key) => ...map to components...
```

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+// Place your settings in this file to overwrite default and user settings.
```
Comment (avahowell) [inline .vscode/settings.json]: remove and gitignore pls :)

Comment (eddiewang) [reply .vscode/settings.json]: ugh. this is why I need to switch to vim.

Comment (eddiewang) [reply .vscode/settings.json]: do you have instructions for rebasing? I tried to do it this morning with yesterday's PR and could not figure out how to properly squash the file from the commit.

**Comment by eddiewang** [comment]
Hopefully this worked. I did the following:
```
git rm -r .vscode
git --amend
```

**Comment by lukechampine** [comment]
If you mistakenly added a file in the previous commit, you can do:
```go
git rm [file]
git commit --amend --no-edit
```
If the commit is "further away" (two or more commits ago), you need to rebase. The quickest way is:
```bash
git rm [file]
git commit --fixup [commit hash]
git rebase -i [commit hash]^ # don't forget the ^
```
This will drop you into an editor, which should show a "fixup" commit directly following the bad commit. Just exit the editor and it should apply everything cleanly.

**Comment by eddiewang** [comment]
I followed @lukechampine 's instructions and believe everything's good to go now. Had to manually apply fixup in interactive mode but I think it did the trick. Also fixed the conflicts from the new fetchData actions added to master.

**Comment by avahowell** [comment]
Can you provide screenshots for this change?

**Comment by eddiewang** [comment]
image

**Comment by eddiewang** [comment]
Nothing pretty...minimal styling - just a simple input box. Used flex layout to quickly push the two elements apart. Had to edit out some priv info.

group

**Comment by avahowell** [comment]
I think this would probably look better if the checkbox were under the `Recent Transaction` header, and had some padding between the box and the label. Also, I think the checkbox should probably default to being checked (set the value of `filter` to true in `initialState`).

**Comment by eddiewang** [comment]
@johnathanhowell what do you think about centering the header text back to orig state? can reduce the default h2 padding if we want to remove the space between the two elements.
image

**Comment by avahowell** [comment]
yeah I think removing the h2 padding would improve the look of this change, ideally to the point where the height of the 'Recent Transactions' banner is the same as it was originally

**Comment by eddiewang** [comment]
@johnathanhowell I changed the padding down to 3px top and bottom.

**Comment by eddiewang** [comment]
@johnathanhowell not sure why travis is failing...

**Comment by avahowell** [comment]
looks like we missed a case where a trailing comma was missing, if you rebase to master `git rebase master` it should pass. Also, can you provide an updated screenshot of how this looks after changing the margins?

**Comment by lukechampine** [comment]
Looks like the `.vscode` file is still there. Run:
```
git rm .vscode/settings.json
git commit --fixup 02f7a2
git rebase -i 02f7a2^
git push --force origin zero-transaction-filter
```

**Comment by eddiewang** [comment]
image

**Comment by eddiewang** [comment]
@johnathanhowell @lukechampine thank you so much for your help on this. I learned a lot about rebasing (hopefully this is less of a headache for future commits). I will post the picture soon... just replaced my macbook and need to resync my node in order to see the wallet page.

**Comment by lukechampine** [comment]
The `fixup!` commit should be squashed into `e2f2261`. If you do `git rebase -i e2f2261^` it should automatically squash them.

At this point it may be easier to squash the entire history. @johnathanhowell, when you merge this, squash it first using the drop-down arrow on the merge button.

**Comment by eddiewang** [comment]
For some reason @lukechampine `git rebase -i e2f2261^` doesn't automatically squash it. Brings me into vim but I went ahead and renamed the `pick` to `squash`  and I think that does the same thing.

**Comment by eddiewang** [comment]
@johnathanhowell 
image

File: plugins/Wallet/js/sagas/wallet.js [hunk 4]
```diff
@@ -32,7 +32,6 @@ function *getLockStatusSaga() {
 		} else {
 			yield put(actions.setUnencrypted())
 		}
-		yield put(actions.setRescanning(response.rescanning))
```
Comment (avahowell) [inline plugins/Wallet/js/sagas/wallet.js]: looks like this snuck in (this line should definitely not be removed)

**Comment by eddiewang** [comment]
yikes. i ran a `git commit --amend` so hopefully that's the reason tests were failing.

**Comment by avahowell** [comment]
This looks good to me, but it has merge conflicts that must be resolved.

**Comment by eddiewang** [comment]
@johnathanhowell with the master branch? seems to have merged fine.

**Comment by eddiewang** [comment]
nvm, you were right. weird. i fetched upstream and merged master before pushing so confused why the conflicts didn't show up earlier.

**Pull request merged**